examples/models/llava/README.md

```diff
@@ -1,20 +1,88 @@
+LLaVA is the first multi-modal LLM ExecuTorch supports. In this directory, we
+- Host a model definition for LLaVA.
+- Demonstrate how to export [LLavA](https://github.com/haotian-liu/LLaVA) multimodal model to a .pte file.
+- Provide a C++ runner that loads the .pte file, the tokenizer and an image, then generate responses based on user prompt.
 
 ## Instructions
-Note that this folder does not host the pretrained LLava model.
-- To have Llava available, follow the [Install instructions](https://github.com/haotian-liu/LLaVA?tab=readme-ov-file#install) in the LLava github. Follow the licence in the specific repo when using L
-- Since the pytorch model version may not be updated, `cd executorch`, run `./install_requirements.sh`.
-- If there is numpy compatibility issue, run `pip install bitsandbytes -I`.
-- Alternatively, run `examples/models/llava_encoder/install_requirements.sh`, to replace the steps above.
-- Run `python3 -m examples.portable.scripts.export --model_name="llava_encoder"`. The llava_encoder.pte file will be generated.
-- Run `./cmake-out/executor_runner --model_path ./llava_encoder.pte` to verify the exported model with ExecuTorch runtime with portable kernels. Note that the portable kernels are not performance optimized. Please refer to other examples like those in llama2 folder for optimization.
-
-## TODO
-- Write the pipeline in cpp
-  - Have image and text prompts as inputs.
-  - Call image processing functions to preprocess the image tensor.
-  - Load the llava_encoder.pte model, run it using the image tensor.
-  - The output of the encoder can be combined with the prompt, as inputs to the llama model. Call functions in llama_runner.cpp to run the llama model and get outputs. The ExecuTorch end to end flow for the llama model is located at `examples/models/llama2`.
+### Export .pte & other artifacts
+
+Run the following command to generate `llava.pte`, `tokenizer.bin` and an image tensor (serialized in TorchScript) `image.pt`.
+
+Prerequisite: run `install_requirements.sh` to install ExecuTorch and run `examples/models/llava/install_requirements.sh` to install dependencies.
+
+```bash
+python -m executorch.examples.models.llava.export_llava --pte-name llava.pte --with-artifacts
+```
+
+Currently the whole export process takes about 6 minutes. We also provide a small test util to verify the correctness of the exported .pte file. Just run:
+
+```bash
+python -m executorch.examples.models.llava.test.test_pte llava.pte
+```
+
+If everything works correctly it should give you some meaningful response such as:
+
+
+
+### Build C++ runner
+
+Run the following cmake commands from `executorch/`:
+
+```bash
+# build libraries
+cmake                                               \
+    -DCMAKE_INSTALL_PREFIX=cmake-out                \
+    -DCMAKE_BUILD_TYPE=Debug                        \
```

Comment:
Any specific reason for debug? Would it affect the runtime speed?

Reply:
Need some testing on release build.